backend/src/api/common.ts

```diff
@@ -71,4 +71,63 @@ export class Common {
        }, ms);
     });
   }
+
+  static setRelativesAndGetCpfpInfo(tx: TransactionExtended, memPool: { [txid: string]: TransactionExtended }): CpfpInfo {
+    const parents = this.findAllParents(tx, memPool);
+
+    let totalWeight = tx.weight + parents.reduce((prev, val) => prev + val.weight, 0);
+    let totalFees = tx.fee + parents.reduce((prev, val) => prev + val.fee, 0);
+
+    tx.ancestors = parents
+      .map((t) => {
+        return {
+          txid: t.txid,
+          weight: t.weight,
+          fee: t.fee,
+        };
+      });
+
+    // Add high (high fee) decendant weight and fees
+    if (tx.bestDescendant) {
+      totalWeight += tx.bestDescendant.weight;
+      totalFees += tx.bestDescendant.fee;
+    }
+
+    tx.effectiveFeePerVsize = totalFees / (totalWeight / 4);
+    tx.cpfpChecked = true;
+
+    return {
+      ancestors: tx.ancestors,
+      bestDescendant: tx.bestDescendant || null,
+    };
+  }
```

Comment:
My sincerest apologies for being nosy, I'm just a big CPFP nerd and fan of mempool. Is this your algorithm for finding the effective feerate of transactions?

Reply:
@glozow Correct. This is my humble CPFP algorithm. If you know ways of improvement it's highly appreciated. The trick is to have an algorithm is also quick since the mempool and all the CPFP numbers are recalculated every ~2 second.

Reply:
The most accurate would be `max(basefeerate, highest ancestorfeerate of all descendants)` because the mining code sorts by highest ancestorfeerate and then puts all of the transaction's direct ancestors in before it. I'm not familiar with the code / know how long this would take but if you're able to query the node, an RPC call `getmempoolentry(txid)` and `res['fees']['ancestor']/res['vsize']` would give you the ancestor feerate.

Reply:
The problem is, it would take too long time to fetch a `getmempoolentry` on all the mempool transactions,, every 2 second. So it has to be calculated naively in the local copy of our NodeJS backend in-memory mempool.

Reply:
Yeah fosho, that makes sense.